numba/cuda/compiler.py

```diff
@@ -831,6 +851,29 @@ def __init__(self, func, sigs, targetoptions):
 
         self.typingctx = CUDATargetDesc.typingctx
 
+        self._tm = default_type_manager
+
+        pysig = utils.pysignature(py_func)
+        arg_count = len(pysig.parameters)
+        argnames = tuple(pysig.parameters)
+        default_values = self.py_func.__defaults__ or ()
+        defargs = tuple(OmittedArg(val) for val in default_values)
+        can_fallback = False # CUDA cannot fallback to object mode
+
+        try:
+            lastarg = list(pysig.parameters.values())[-1]
+        except IndexError:
+            has_stararg = False
+        else:
+            has_stararg = lastarg.kind == lastarg.VAR_POSITIONAL
+
+        exact_match_required = False
+
+        _dispatcher.Dispatcher.__init__(self, self._tm.get_pointer(),
+                                        arg_count, self._fold_args, argnames,
+                                        defargs, can_fallback, has_stararg,
+                                        exact_match_required)
```

Comment:
Whilst some of this is just setting things to the default values in the `_DispatcherBase` its probably a good idea to keep this as-is such that all these items are considered when doing the next refactoring step.

Reply:
As `nu,mba.cuda.Dispatcher` is not yet inheriting from `_DispatcherBase`, I need to set these things here. (Am I understanding your remark correctly?)

Reply:
yes, this is fine.